TimberAPI/SpecificationSystem/TimberApiSpecificationService.cs

```diff
@@ -71,6 +79,60 @@ public IEnumerable<T> GetSpecifications<T>(IObjectSerializer<T> serializer)
             }
         }
 
+        /// <summary>
+        /// Generates Custom original BuildingSpecification jsons for
+        /// all buildings in the game
+        /// </summary>
+        /// <returns></returns>
+        private IEnumerable<TextAsset> GenerateBuildingSpecifications()
+        {
+            var buildings = new List<TextAsset>();
+            var buildingComponents = Resources.LoadAll<Timberborn.Buildings.Building>(_buildingsPath);
+
+            foreach (var component in buildingComponents)
+            {
+                var manufactory = component.GetComponent<Manufactory>();
+                List<Recipe> recipes = new List<Recipe>();
+                if (manufactory != null)
+                {
+                    recipes = manufactory.ProductionRecipeIds
+                                         .Select(x => new Recipe(x))
+                                         .ToList();
+                }
+                var build = new Buildings.Building(component.ScienceCost,
+                                                   component.BuildingCost
+                                                            .Select(x => new BuildingCost(x.GoodId, x.Amount))
+                                                            .ToList());
+                var mechanicalNode = component.GetComponent<Timberborn.MechanicalSystem.MechanicalNodeSpecification>();
+                MechanicalNode mechNode = null;
+                if (mechanicalNode != null)
+                {
+                    mechNode = new MechanicalNode(mechanicalNode.PowerInput, mechanicalNode.PowerOutput);
+                }
+
+                var buildingSpec = new BuildingSpecification(component.name,
+                                                             recipes,
+                                                             build,
+                                                             mechNode);
+
+                var bar = new JsonSerializer();
```

Comment:
what is going on here

Reply:
I agree, I cannot understand :D. We could leave it for now OR try already adding a whole new system for this kind of generation that can be seperated within classes and SOLID. Maybe like how the singleton attributes are done with bindito? Since this isn't going to be the last generation we have to do if we add more things.